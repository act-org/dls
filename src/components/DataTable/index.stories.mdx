<!-- @prettier -->

import {
  ArgsTable,
  Canvas,
  Description,
  Meta,
  Preview,
  Source,
  Story,
} from '@storybook/addon-docs';

import { argTypes, Template } from './index.stories.template';
import { DataTable, DataTableProps } from '.';

<Meta
  title="V3/Organisms/DataTable"
  component={DataTable}
  argTypes={argTypes}
/>

<Description of={DataTable} />

## Color: Primary

<Canvas>
  <Story
    name="Color: Primary"
    args={{
      color: 'primary',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Color: Secondary

<Canvas>
  <Story
    name="Color: Secondary"
    args={{
      color: 'secondary',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Playground

<Canvas isColumn>
  <Source language="jsx" story="Playground" />
  <Story name="Playground" args={{}}>
    {Template.bind({})}
  </Story>
  <ArgsTable story="Playground" />
</Canvas>
