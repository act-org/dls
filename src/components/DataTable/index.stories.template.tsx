/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { Story } from '@storybook/react/types-6-0';
import { isNumber } from 'lodash';
import moment from 'moment';
import { useState } from 'react';

import SORT_DIRECTION_TYPES from '@actinc/dls/constants/SORT_DIRECTION_TYPES';
import { Playground } from '@actinc/dls/helpers/playground';
import sort from '@actinc/dls/helpers/sort';
import { SortObject } from '@actinc/dls/types';

import { DataTable, DataTableProps } from '.';

interface Item {
  id: string;
  name: string;
  updatedAt: Date;
  fieldA: string;
  fieldB: string;
  fieldC: string;
}

export const Template: Story<DataTableProps<Item>> = ({
  limit: limitProps,
  offset: offsetProps,
  totalCount,
  ...args
}: DataTableProps<Item>) => {
  const [limit, setLimit] = useState<number | undefined>(limitProps);
  const [offset, setOffset] = useState<number | undefined>(offsetProps);
  const [sortObject, setSortObject] = useState<SortObject>({
    sortBy: 'id',
    sortDirection: SORT_DIRECTION_TYPES.ASCENDING,
  });

  let items: Item[] = [...Array(totalCount || 0)].map((_, i): any => ({
    fieldA: `Field A${i + 1}`,
    fieldB: `Field B${i + 1}`,
    fieldC: `Field C${i + 1}`,
    id: i + 1,
    name: `Item ${i + 1}`,
    updatedAt: moment()
      .subtract(2, 'year')
      .subtract(i + 1, 'day')
      .toDate(),
  }));

  if (isNumber(limit) && isNumber(offset)) {
    items = items.slice(offset, offset + limit);
  }

  // sort items
  items = items.sort(sort<Item>(sortObject));

  return (
    <DataTable<Item>
      {...args}
      columns={[
        {
          label: 'ID',
          renderValue: (i: Item): string => i.id,
          sortBy: 'id',
          style: {
            width: 50,
          },
        },
        {
          label: 'Name',
          renderValue: (i: Item): string => i.name,
          sortBy: 'name',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field A',
          renderValue: (i: Item): string => i.fieldA,
          sortBy: 'fieldA',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field B',
          renderValue: (i: Item): string => i.fieldB,
          sortBy: 'fieldB',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field C',
          renderValue: (i: Item): string => i.fieldC,
          sortBy: 'fieldC',
          style: {
            width: 100,
          },
        },
        {
          label: 'Updated',
          renderValue: (i: Item): string =>
            moment(new Date(i.updatedAt)).fromNow(),
          sortBy: 'updatedAt',
          style: {
            width: 110,
          },
        },
      ]}
      currentSortObject={sortObject}
      items={items}
      limit={limit}
      offset={offset}
      onChangeLimit={setLimit}
      onChangeOffset={setOffset}
      onChangeSort={setSortObject}
      totalCount={totalCount}
    />
  );
};

export const argTypes = Playground(
  {
    color: { type: 'string' },
  },
  DataTable,
);
