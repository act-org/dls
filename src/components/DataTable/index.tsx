/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import * as React from 'react';
import { Table, TableBody, TableHead, TableRow } from '@material-ui/core';

import { EmptyState, EmptyStateProps } from '~/components/EmptyState';
import { SortObject } from '~/types';

import TableCellBody from './TableCellBody';
import TableCellHead from './TableCellHead';
import TableContainer from './TableContainer';

import useStyles from './styles';

interface Column<T> {
  renderValue: (item: T) => any;
  label?: string;
  sortBy?: string;
  style?: React.CSSProperties;
}

export interface DataTableProps<T> {
  color?: 'primary' | 'secondary';
  columns: Column<T>[];
  currentSortObject: SortObject;
  emptyStateProps?: EmptyStateProps;
  items: T[];
  onChangeSort: (sortObject: SortObject) => void;
  RowWrapper?: (
    item: T,
    children: React.ReactElement<any>,
  ) => React.ReactElement<any>;
}

export const DataTable = <T,>({
  color,
  columns,
  currentSortObject,
  emptyStateProps,
  items,
  onChangeSort,
  RowWrapper,
}: DataTableProps<T>): React.ReactElement<any> => {
  const classes = useStyles();

  return (
    <TableContainer>
      <Table>
        <TableHead>
          <TableRow>
            {columns.map(
              (column, i): React.ReactElement<any> => (
                <TableCellHead
                  color={color}
                  currentSortObject={currentSortObject}
                  key={column.label || i}
                  onChangeSort={onChangeSort}
                  sortBy={column.sortBy}
                  style={column.style}
                >
                  {column.label || ''}
                </TableCellHead>
              ),
            )}
          </TableRow>
        </TableHead>

        <TableBody>
          {items.map((item, i): React.ReactElement<any> => {
            /* eslint-disable react/no-array-index-key */
            const children = (
              <TableRow hover key={i}>
                {columns.map(
                  (column, y): React.ReactElement<any> => (
                    <TableCellBody key={column.label || y} style={column.style}>
                      {column.renderValue(item)}
                    </TableCellBody>
                  ),
                )}
              </TableRow>
            );
            /* eslint-enable react/no-array-index-key */

            if (RowWrapper) {
              return RowWrapper(item, children);
            }

            return children;
          })}
        </TableBody>
      </Table>

      {items.length === 0 && emptyStateProps && (
        <div className={classes.emptyStateContainer}>
          <EmptyState description="No Results Found" {...emptyStateProps} />
        </div>
      )}
    </TableContainer>
  );
};

DataTable.defaultProps = {
  color: undefined,
  RowWrapper: undefined,
};

export default DataTable;
