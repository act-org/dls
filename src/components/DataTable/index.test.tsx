/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

;
import { noop } from 'lodash';

import { snapshot } from '~/helpers/test';

import DataTable from '.';

interface Item {
  id: string;
  name: string;
  fieldA: string;
  fieldB: string;
  fieldC: string;
}

describe('DataTable', () => {
  const Component = (
    <DataTable
      columns={[
        {
          label: 'ID',
          renderValue: (i: Item): string => i.id,
          sortBy: 'id',
          style: {
            width: 50,
          },
        },
        {
          label: 'Name',
          renderValue: (i: Item): string => i.name,
          sortBy: 'name',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field A',
          renderValue: (i: Item): string => i.fieldA,
          sortBy: 'fieldA',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field B',
          renderValue: (i: Item): string => i.fieldB,
          sortBy: 'fieldB',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field C',
          renderValue: (i: Item): string => i.fieldC,
          sortBy: 'fieldC',
          style: {
            width: 100,
          },
        },
      ]}
      currentSortObject={{
        sortBy: 'id',
        sortDirection: 'ASCENDING',
      }}
      items={Array(...Array(5)).map((_, i): any => ({
        fieldA: `Field A${i + 1}`,
        fieldB: `Field B${i + 1}`,
        fieldC: `Field C${i + 1}`,
        id: i + 1,
        name: `Item ${i + 1}`,
      }))}
      onChangeSort={noop}
    />
  );
  snapshot(Component);
});
