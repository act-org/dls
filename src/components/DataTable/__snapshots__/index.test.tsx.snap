// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable ACT theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-paperRoot-2 MuiTableContainer-root MuiPaper-elevation0 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRoot-8 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 50px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-6 makeStyles-typographyRoot-9 MuiTypography-body1"
            >
              ID
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-7"
                focusable="false"
                role="button"
                style="color: rgb(255, 255, 255);"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-7"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRoot-11 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-6 makeStyles-typographyRoot-12 MuiTypography-body1"
            >
              Name
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-10"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-10"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRoot-14 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-6 makeStyles-typographyRoot-15 MuiTypography-body1"
            >
              Field A
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-13"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-13"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRoot-17 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-6 makeStyles-typographyRoot-18 MuiTypography-body1"
            >
              Field B
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-16"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-16"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRoot-20 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-6 makeStyles-typographyRoot-21 MuiTypography-body1"
            >
              Field C
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-19"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4 makeStyles-sortIconRoot-19"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C1
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C2
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C3
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C4
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-22 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C5
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable ACT_ET theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-paperRoot-24 MuiTableContainer-root MuiPaper-elevation0 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-27 makeStyles-tableCellRoot-30 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 50px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-28 makeStyles-typographyRoot-31 MuiTypography-body1"
            >
              ID
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-25 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-29"
                focusable="false"
                role="button"
                style="color: rgb(255, 255, 255);"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-29"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-27 makeStyles-tableCellRoot-33 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-28 makeStyles-typographyRoot-34 MuiTypography-body1"
            >
              Name
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-25 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-32"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-32"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-27 makeStyles-tableCellRoot-36 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-28 makeStyles-typographyRoot-37 MuiTypography-body1"
            >
              Field A
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-25 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-35"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-35"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-27 makeStyles-tableCellRoot-39 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-28 makeStyles-typographyRoot-40 MuiTypography-body1"
            >
              Field B
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-25 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-38"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-38"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-27 makeStyles-tableCellRoot-42 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-28 makeStyles-typographyRoot-43 MuiTypography-body1"
            >
              Field C
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-25 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-41"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-26 makeStyles-sortIconRoot-41"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </td>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C1
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C2
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C3
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C4
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 50px;"
          >
            5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Item 5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field A5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field B5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-44 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            style="width: 100px;"
          >
            Field C5
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
